src/runtime_src/core/edge/sw_em/generic_pcie_hal2/shim.cxx

```diff
@@ -1855,6 +1857,63 @@ int CpuemShim::xclCloseContext(const uuid_t xclbinId, unsigned int ipIndex) cons
   return 0;
 }
 
+
+int CpuemShim::xclIPName2Index(const char *name)
+{ 
+  std::string errmsg;
+  const uint64_t bad_addr = 0xffffffffffffffff;
+  char* ipLayoutbuf = nullptr;
```

Comment:
`auto ipLayoutbuf = mCoreDevice->get_axlf_section(IP_LAYOUT);`

Reply:
used mCoreDevice to get IP_LAYOUT section.